cmd/collectors/ems/ems.go

```diff
@@ -363,23 +417,31 @@ func (e *Ems) PollData() (map[string]*matrix.Matrix, error) {
 	timeFilter := e.getTimeStampFilter()
 	filter := append(e.Filter, timeFilter)
 
-	// collect all event names
-	var names []string
-	for key := range e.emsProp {
-		names = append(names, key)
-	}
-
-	// Split names into batches
-	batch := e.batchSize
-
-	for i := 0; i < len(names); i += batch {
-		j := i + batch
-		if j > len(names) {
-			j = len(names)
+	// build hrefs based on urlsize
+	var hrefs []string
+	start := 0
+	for end := 0; end < len(e.eventNames); end += 1 {
+		h := e.getHref(e.eventNames[start:end], filter)
+		if len(h) > e.maxUrlSize {
+			if end == 0 {
+				return nil, fmt.Errorf("url size is too small to form queries: %d", e.maxUrlSize)
```

Comment:
logs quite confusing, I believe url size is large even without adding ems name, right ?